### DISCUSSION TITLE
Exporting ocr output onto a txt file

### DISCUSSION THREAD

**Discussion by orlandito24** [mindee/doctr#1314] [Q&A]
Hi, I'm trying to find a way to export the results of a model onto a text file, while retaining the structure of original pdf/image/url etc.

I've tried using `render()` but that is just pushing any found words to the left margin. If there is a word in the middle of the page on my pdf, I would want the resulting string to have whitespace padding up to where that word begins.

**Comment by charlesmindee** [comment]
Hi @orlandito24, thanks for your interest in doctr!
For the moment there is no such a feature, you can keep the line order but the indent is not preserved. 
However, we have a [synthetize_page](https://github.com/mindee/doctr/blob/19f101cd300f9d978213e4c5d30944fafdbe3700/doctr/utils/visualization.py#L278) function which draws the words on a blank page where they are detected. It is exported as an image and not a text file, but maybe you can take inspiration from this code to convert the raw response to a structured text file (writing the file line by line and considering each pixel between 2 boxes to be a space " ")
I hope it can help!
